RTProtocol.cs

```diff
@@ -423,31 +452,34 @@ public int ReceiveRTPacket(out PacketType packetType, bool skipEvents = true, in
                     while (receivedTotal < frameSize)
                     {
                         // As long as we haven't received enough data, wait for more
-                        received = mNetwork.Receive(ref data, receivedTotal, frameSize - receivedTotal, false, -1);
-                        if (received <= 0)
+                        response = mNetwork.Receive(ref data, receivedTotal, frameSize - receivedTotal, false, -1);
+                        if (response == Response.ResponseType.timeout)
                         {
-                            if (!mNetwork.IsConnected())
-                            {
-                                mErrorString = "Disconnected from server.";
-                            }
-                            else
-                            {
-                                mErrorString = "Socket Error.";
-                            }
-                            return -1;
+                            return ReceiveResponseType.timeout; // Receive timeout
+                        }
```

Comment:
Remove this timeout check since it can never happen. The Receive above is blocking ([A -1 value indicates an infinite time-out.](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.socket.select?view=net-5.0))

Reply:
I changed to non-blocking receive with 5 sec timeout when waiting for consecutive parts. And return an error on timeout.
With blocking, we could potentially get a lock if we don't get all parts.

Reply:
Ok good!